### PR TITLE
misc cleanups

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: include/exiv2/types.hpp [hunk 1]
```diff
@@ -461,11 +461,7 @@ namespace Exiv2 {
         return rc == src + N ? nullptr : rc;
     }
 
-    //! Template used in the COUNTOF macro to determine the size of an array
-    template <typename T, int N>
-    char (&sizer(T (&)[N]))[N];
-//! Macro to determine the size of an array
-#define EXV_COUNTOF(a) (sizeof(Exiv2::sizer(a)))
+#define EXV_COUNTOF(a) std::size(a)
```
Comment (neheb) [inline include/exiv2/types.hpp:464]: I kept this in here just in case. Do we want to remove it?

Comment (piponazo) [reply include/exiv2/types.hpp:464]: If you have made all the replacements to use directly `std::size()`, I would remove it (now this line should be Dead Code)

Comment (neheb) [reply include/exiv2/types.hpp]: Done.

**Review by piponazo — APPROVED** [review]
LGTM, thanks! ❤️

**Pull request merged**